### ISSUE TITLE
Drag Drop Files between folders

### ISSUE THREAD

**Issue by d3ep4k** [alexcurtis/react-treebeard#18]
Hello,

Firstly I would like to congratulate for making such a good treeview. 
How can we use the drag and drop to move files around.

Thanks,
Deepak

**Comment by alexcurtis** [comment]
@d3ep4k Yes, this would be a pretty cool feature. I'll experiment and add it into the next major release

**Comment by hannupekka** [comment]
Any progress with this? :)

**Comment by sebastianconcept** [comment]
@alexcurtis I'd choose this component if it would drag & drop

**Comment by 029xue** [comment]
waiting for this feature as well.

**Comment by coolguy001tv** [comment]
waiting +1

**Comment by radman-rt** [comment]
waiting +1

**Comment by shindigira** [comment]
@alexcurtis I modified treebeard with drag and drop. Would you like me to do a PR with this?
![treebeard dnd](https://user-images.githubusercontent.com/13324863/31108598-4d86b3ba-a7c8-11e7-9ad7-a49a0e730b81.png)

**Comment by MichaelDeBoey** [comment]
@shindigira You can always make a PR, so we can review it and maybe add it afterwards. :-)

**Comment by asifmom** [comment]
@shindigira any update on the PR?

**Comment by firetiger77** [comment]
@shindigira I'm looking into adding DnD capabilities using react-dnd, but if you've already done so, I'd love to leverage what you've got.

**Comment by codecounselor** [comment]
I'm evaluating tree components to pair up with a [react-table](https://github.com/react-tools/react-table).  What I need is DnD support within the tree and from the tree to the table.  

I really like the design decision of react-table to allow component overrides
https://github.com/react-tools/react-table#component-overrides

It looks like this project already does this to some extent.  Is there any opposition to expand this to allow the TreeNode, etc. to be overridden in a manner like react-table?

Having DnD  built into the tree would certainly be nice as well.  But I'm guessing others have the same requirements I'm faced with.

**Comment by shindigira** [comment]
@asifmom @firetiger77 Apologies for not responding for this many moons ago. Would you still like me to PR this? I have it still as WIP and not flushed out.

**Comment by mismith** [comment]
Same story: I'm looking to add dnd support, and I can't seem to find any existing plugins or examples of react-treebeard using react-dnd. Does anyone have any links they could share to save me writing it from scratch myself?

**Comment by maximilianoforlenza** [comment]
Hi! I'm going to start work this issue.

**Comment by mismith** [comment]
Any luck @maximilianoforlenza ?

**Comment by durdevic** [comment]
@maximilianoforlenza  These guys made it, maybe you can see how they handle it https://github.com/react-component/tree 👌

**Comment by kombuchafox** [comment]
I am a developer from samsara.com . We're interested in using this component and want to give back to the community. I can help if you need it  @maximilianoforlenza

**Comment by piotrkochan** [comment]
any progress? :)

**Comment by kelvinkoko** [comment]
any update about this? i am interested to help if necessary
